### DISCUSSION TITLE
Runtime Error: illegal hardware exception

### DISCUSSION THREAD

**Discussion by BigDataWriter** [ozel/DIY_particle_detector#49] [Q&A] [answered]
My conda environment is using python 3.9.5, which appears to be too new to run the code for the particle detector.  What version of python 3.x are other people using successfully?

/Users/robjustice/Downloads/DIY_particle_detector/data_recording_software/./pulse_recorder.py:147: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.df = self.df.append(pulse, ignore_index=True,sort=False)
zsh: illegal hardware instruction  python ./pulse_recorder.py

**Comment by ozel** [comment, marked as answer]
The shown python warning is really just that, a warning. It will not impact the running of the script.
I have developed and tested this extensively with python 3.6 & 3.7. Can test it in the coming days on 3.9. if your problem persists.

Googling your error ```zsh: illegal hardware instruction``` sounds like some general python issue on Apple M1 systems. Do you use a recent Mac featuring an ARM processor?
This error seems generally related to specific Python modules that may need adaptation for ARM/Apple M1 systems. You could try manually importing all the modules listed at the beginning of the script inside an interactive python session and check which one fails.

**Comment by BigDataWriter** [comment]
Oliver,

Thank you for your response!  I’m able to run the software now using a python 3.6 environment.

Best,
Rob

**Reply by ozel** [reply]
great! could you still let me know more about your environment/operating system? Is it a recent Mac with M1 chip?

**Comment by BigDataWriter** [comment]
Oliver,

Hardware specs in the image below.



As stated earlier, software environment was python 3.6.  To clarify, I successfully ran and used the software, but I didn’t succeed in getting the gun to run, so my setup isn’t yet ready for a potential documentation update of supported, recommended configurations.

Rob
